### DISCUSSION TITLE
Will ixgbe(vf) driver present in next major release (23.05) ?

### DISCUSSION THREAD

**Discussion by miyatsu** [immortalwrt/immortalwrt#958] [General]
The ixgbe(vf) vendor driver was added in 21.02 branch in commit 34b260657089e4993c88298c57e10c50b4066011, will this change be cherry-picked to 23.05 branch?

**Comment by 1715173329** [comment]
You should simply install in-tree kmod-ixgbe and kmod-ixgbevf module

**Reply by miyatsu** [reply]
I just hope the ixgbevf driver will be present in x86 firmware out of the box, without any opkg install steps in next 23.05 release.

Consider this:

For immortalwrt running in vm with only SR-IOV device assigned to it, there is impossible to connect to the Internet to install packages without ixgbevf driver.

To solve this issue we have to assign none SR-IOV device (vmxnet3/e1000/virtio-net/etc.) first to the vm and install ixgbevf driver via opkg, then replace the none SR-IOV device with SR-IOV enabled device after shutdown the VM. Or we can simply make the `ixgbevf` as built-in module for x86 firmware like you did in 21.02 branch.

**Comment by miyatsu** [comment]
Clean installation OpenWRT in VM environment is pain in the ass, it requires manual configuration (as you mationed install `kmod-ixgbevf` via opkg) before boot up with SR-IOV device. For ImmortalWRT, built-in `kmod-ixgbevf` will save me a lot of manual configuration time in last 21.02 relase. I was hope this driver can be usable out of the box in next major 23.05 relase.

**Comment by 1715173329** [comment]
Fine, will add them to default packages later.

**Reply by 1715173329** [reply]
btw the [_firmware selector_](https://firmware-selector.immortalwrt.org/) allows you to custom the pre-installed packages

**Reply by miyatsu** [reply]
Wow, I'm not aware of this feature. I will try it when I got free time.👍

**Comment by 1715173329** [comment]
https://github.com/immortalwrt/immortalwrt/commit/b4ef8831a92a7d39253455d4a44c2ee5317e131c
it's in the default set already.
